src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -72,27 +72,23 @@ bool ChaseMovementGenerator<T>::DoUpdate(T* owner, uint32 time_diff)
         return false;
 
     Creature* cOwner = owner->ToCreature();
-    bool isStoppedBecauseOfCasting = cOwner && cOwner->IsMovementPreventedByCasting();
 
+    r_leashExtensionTimer.Update(time_diff);
     // the owner might be unable to move (rooted or casting), or we have lost the target, pause movement
-    if (owner->HasUnitState(UNIT_STATE_NOT_MOVE) || HasLostTarget(owner) || isStoppedBecauseOfCasting)
+    if (owner->HasUnitState(UNIT_STATE_NOT_MOVE) || HasLostTarget(owner) || (cOwner && cOwner->IsMovementPreventedByCasting()))
     {
         owner->StopMoving();
         _lastTargetPosition.reset();
         if (cOwner)
         {
-            if (isStoppedBecauseOfCasting)
+            if (r_leashExtensionTimer.Passed())
             {
-                // Don't reset leash timer if it's a spell like Shoot with a short cast time.
-                /// @todo: Research how it should actually work.
-                Spell *spell = cOwner->GetFirstCurrentCastingSpell();
-                bool spellHasLongCast = spell && spell->GetCastTime() > 1 * SECOND * IN_MILLISECONDS;
-                if (spellHasLongCast)
+                if (!i_target->isMoving())
```

Comment:
In this case It will never reset timer if a player constantly moving, right? I don't think it's right.

Reply:
Using the method of checking the creature cast time does not properly leave the fight.

Reply:
Sorry, can you provide an example scenario? I just had the impression that it worked fine for me before.

Reply:
Okay, wait a minute

Reply:
> Sorry, can you provide an example scenario? I just had the impression that it worked fine for me before.

.go c id 5617
![Uploading image.png…]()

Reply:
The issue we had before was that creatures would reset before they manage to cast a single spell if the leash timer is too short

Reply:
> The issue we had before was that creatures would reset before they manage to cast a single spell if the leash timer is too short

Yes, the PR fixed that bug.
I've put the PR on the server and testing it